File: lotto.js
/*

1부터 45까지 숫자 중 6개 비복원 추출
> [2,5,34,12,23,11]

(힌트) lodash 활용하면 쉽다

*/

const _ = require('lodash')

// 1. 1~45까지 담긴 동 생성

const numbers = _.range(1,46)

// 2, 랜덤으로 6개 출력

console.log(_.sampleSize(numbers,6))

// 3. 정렬

console.log(_.sortBy(_.sampleSize(numbers,6)))


//  winner와 로또를 비교해서 몇개가 일치하는지 그리고 몇등인지 출력
// 6개 ,1등
// 5개, 3등
// 4개, 4등
// 3개, 5등

// const winner = [1,5,19,23,28,42]
// const lotto = _.sortBy(_.sampleSize(numbers,6))

// if (_.isEqual(winner, lotto)) {
//     console.log("6개, 1등");
// }
    
// else if (_.without(winner,lotto) = _.sampleSize = 1) {
//  console.log("5개, 3등");
// }

// else if (_.without(winner,lotto) = _.sampleSize = 2 ) {
//     console.log("4개, 4등");
// }

// else if ( _.without(winner,lotto) = _.sampleSize = 3) {
//     console.log("3개, 5등");
// }
// else   

// console.log("2개, 꽝");


const cnt = _.intersection(lotto,winner).length
console.log 

if  (cnt == 1) {
    rank = 1등입니다}


lotto .forEach(function(num) {
    if (winner.includes(num)) {
        cnt += 1;
    }
})

File: lotto.py
import random

def getLotto():
    winner = [1, 15, 19, 23, 28, 42]

    lotto = sorted(random.sample(range(1, 46), 6))

    cnt = len(set(winner) & set(lotto))

    # 6, 1등
    # 5, 3등
    # 4, 4등
    # 3, 5등

    rank = ''
    if cnt == 6:
        rank = '1등'
    elif cnt == 5:
        rank = '3등'
    elif cnt == 4:
        rank = '4등'
    elif cnt == 3:
        rank = '5등'
    else:
        rank = '꽝'

    print(f'{cnt}, {rank}')


# winner = [1,5,19,23,28,42]

# lotto = sorted(random.sample(range(1,46),6))

# print(set(winner) & set(lotto))

# print(len(set(winner) & set(lotto)))

# cnt = len(set(winner) & set(lotto)

# if cnt == 6:
#     print(len(set(winner) + ", " + "1등입니다"))

# elif cnt ==5:
#     print(len(set(winner) + ", " + "3등입니다"))

# elif  cnt == 4:
#     print(len(set(winner) + ", " + "4등입니다"))

# elif  cnt == 3:
#     print(len(set(winner) + ", " + "5등입니다"))

# else
#     print(len(set(winner) + ", " + "꽝입니다"))